packages/compass-e2e-tests/helpers/compass.ts

```diff
@@ -196,20 +197,17 @@ export class Compass {
 
     // get the app logPath out of electron early in case the app crashes before we
     // close it and load the logs
-    this.logPath = await this.browser.execute(() => {
-      // eslint-disable-next-line @typescript-eslint/no-var-requires
-      return require('electron').ipcRenderer.invoke('compass:logPath');
-    });
-
-    this.userDataPath = await this.browser.execute(() => {
-      // eslint-disable-next-line @typescript-eslint/no-var-requires
-      return require('electron').ipcRenderer.invoke('compass:userDataPath');
-    });
-
-    this.appName = await this.browser.execute(() => {
-      // eslint-disable-next-line @typescript-eslint/no-var-requires
-      return require('electron').ipcRenderer.invoke('compass:appName');
-    });
+    [this.logPath, this.userDataPath, this.appName, this.mainProcessPid] =
+      await this.browser.execute(() => {
+        // eslint-disable-next-line @typescript-eslint/no-var-requires
+        const { ipcRenderer } = require('electron');
+        return Promise.all([
```

Comment:
`executeAsync` seems to be callback-based rather than Promise-based, right?

Reply:
It is. I just assumed that using callback-style with `executeAsync` is the only way to get async result because for `execute` the documentation says the following:

> Inject a snippet of JavaScript into the page for execution in the context of the currently selected frame. The executed script is **assumed to be synchronous** and the result of evaluating the script is returned to the client.

So based on that I thought that `execute` would just try to return a promise without waiting for it to resolve

Reply:
But clearly it worked before already, so this just seems like an undocumented behavior, not an issue in the code, I was just genuinely surprised to see it and didn't notice this before 🙂